### PR TITLE
Added cover letter field and new email types

### PR DESCRIPTION
**Pull request by devalirizwan**

### CODE REVIEW

**Comment by devhassanmahmood** [comment]
[A new field "Cover letter explaining changes" is added to the proposal application Step 1 when the Revision - Round 2 is requested by the staff member](https://trello.com/c/fsI7OxVT/661-a-new-field-cover-letter-is-added-when-the-revision-is-requested-by-the-staff-member)

File: app/views/proposals/_proposal.html.erb [hunk 1]
```diff
@@ -57,6 +57,15 @@
                     <div class="tab-content">
                       <div class="tab-pane <%= active_tab(@proposal, 'one', params[:tab]) %>" id="tab-1" role="tabpanel">
                         <h4 class="tab-title"></h4>
+                        <% if @proposal.revision_requested_2? %>
+                          <div class="mb-3">
+                            <h3>Cover Letter</h3>
+                            <p>
+                              This field explains the changes in proposal.
+                            </p>
+                            <%= f.text_area(:cover_letter, size: '5x5', class:'form-control', value: @proposal&.cover_letter, disabled: action)%>
```
Comment (brentkearney) [inline app/views/proposals/_proposal.html.erb:66]: `, disabled: action) %>`

**Review by brentkearney — CHANGES_REQUESTED** [review]
Please refactor to change static "revision_1", "revision_2", etc, to dynamic revision numbers. Assume there could be any number of revisions. 

Also: please avoid using @instance variables unless they are absolutely necessary. Pass arguments to methods.

File: app/views/proposals/_proposal.html.erb (hunk 1 above)
Comment (brentkearney) [reply app/views/proposals/_proposal.html.erb:66]: (missing space)

**Comment by brentkearney** [comment]
Also, refactoring methods to reduce size & complexity would be helpful, if there is time.

https://app.codacy.com/gh/birs-math/proposals/pullRequest?prid=8354348

**Comment by devhassanmahmood** [comment]
[Add a new status 'Shortlisted' to the proposal workflow](https://trello.com/c/J75jUgv5/663-add-a-new-status-shortlisted-to-the-proposal-workflow)

**Pull request merged**